notebook/notebook/notebook.py

```diff
@@ -88,72 +96,80 @@ def web_authenticated_workshop_guest_only(fun):
-        image = userdata['workshop_image']
-    else:
-        image = DEFAULT_WORKER_IMAGE
-    volumes = [
-        kube.client.V1Volume(
-            name='deploy-config',
-            secret=kube.client.V1SecretVolumeSource(
-                secret_name='deploy-config'))
-    ]
-    volume_mounts = [
-        kube.client.V1VolumeMount(
-            mount_path='/deploy-config',
-            name='deploy-config',
-            read_only=True)
-    ]
 
-    user_id = userdata['id']
-    ksa_name = userdata.get('ksa_name')
+    if service == 'notebook':
+        service_account_name = userdata.get('ksa_name')
 
-    bucket = userdata.get('bucket_name')
-    if bucket is not None:
+        bucket = userdata['bucket_name']
         command.append(f'--GoogleStorageContentManager.default_path="{bucket}"')
 
-    gsa_key_secret_name = userdata.get('gsa_key_secret_name')
-    if gsa_key_secret_name is not None:
-        volumes.append(
+        image = DEFAULT_WORKER_IMAGE
+
+        env = [kube.client.V1EnvVar(name='HAIL_DEPLOY_CONFIG_FILE',
+                                    value='/deploy-config/deploy-config.json')]
+
+        jwt_secret_name = userdata['jwt_secret_name']
+        gsa_key_secret_name = userdata['gsa_key_secret_name']
+        volumes = [
             kube.client.V1Volume(
-                name='gsa-key',
+                name='deploy-config',
                 secret=kube.client.V1SecretVolumeSource(
-                    secret_name=gsa_key_secret_name)))
-        volume_mounts.append(
-            kube.client.V1VolumeMount(
-                mount_path='/gsa-key',
+                    secret_name='deploy-config')),
+            kube.client.V1Volume(
                 name='gsa-key',
-                read_only=True))
-
-    jwt_secret_name = userdata.get('jwt_secret_name')
-    if jwt_secret_name is not None:
-        volumes.append(
+                secret=kube.client.V1SecretVolumeSource(
+                    secret_name=gsa_key_secret_name)),
             kube.client.V1Volume(
                 name='user-tokens',
                 secret=kube.client.V1SecretVolumeSource(
-                    secret_name=jwt_secret_name)))
-        volume_mounts.append(
+                    secret_name=jwt_secret_name))
+        ]
+        volume_mounts = [
+            kube.client.V1VolumeMount(
+                mount_path='/deploy-config',
+                name='deploy-config',
+                read_only=True),
+            kube.client.V1VolumeMount(
+                mount_path='/gsa-key',
+                name='gsa-key',
+                read_only=True),
             kube.client.V1VolumeMount(
                 mount_path='/user-tokens',
                 name='user-tokens',
-                read_only=True))
+                read_only=True)
+        ]
+        resources = kube.client.V1ResourceRequirements(
+            requests={'cpu': '1.601', 'memory': '1.601G'})
+    else:
+        workshop = userdata['workshop']
```

Comment:
Small preference: I prefer having the simpler condition first, easier to track, for me.

Reply:
I don't really care, but I have a preference for the "main" or "common" case first, and we should think of the notebook service as more important than the workshop service (eventually)

Reply:
If you still prefer, I'll change it.

Reply:
No need. I like handling the failure condition first, but it doesn't matter

Reply:
Yeah, I prefer return early first, too.  You don't need a else in that case, tho.